src/v/cluster/tx_topic_manager.cc

```diff
@@ -45,7 +45,101 @@ tx_topic_manager::tx_topic_manager(
   , _segment_size(std::move(segment_size))
   , _retention_duration(std::move(retention_duration)) {}
 
-ss::future<> tx_topic_manager::start() { co_return; }
+ss::future<> tx_topic_manager::start() {
+    co_await do_reconcile_topic_properties();
+
+    _segment_size.watch([this] { reconcile_topic_properties(); });
+    _retention_duration.watch([this] { reconcile_topic_properties(); });
+}
+
+void tx_topic_manager::reconcile_topic_properties() {
+    ssx::spawn_with_gate(
+      _gate, [this] { return do_reconcile_topic_properties(); });
+}
+
+ss::future<> tx_topic_manager::do_reconcile_topic_properties() {
+    /**
+     * We hold mutex to make sure only one instance of reconciliation loop is
+     * active. Properties update evens are asynchronous hence there would be a
+     * possibility to have more than one reconciliation process active.
+     */
+    auto u = _reconciliation_mutex.get_units();
+
+    vlog(txlog.trace, "Reconciling tx manager topic properties");
```

Comment:
should this happen only on one node? (controller leader?), otherwise we will as many updates as # nodes in the cluster?

Reply:
i was thinking about this, the topic update property is idempotent. Those redundant topic updates will not negatively impact the cluster. If we would want to do it only on controller leader we would need additional complexity to check for the leadership changes. Also this operation will be pretty rare.

Reply:
Right, a hack is probably to add some jitter before checking for reconciliation (like voter priority) but thats too much for this code I guess :-)